Add media section (#14)
open-education-hub/methodology: description.md

### Media

The `media` content type refers to audio, visual or audiovisual educational materials.

> Images, videos, recordings, educational video games represent examples of such content.

Reviewer: Nice to think about video games.

Reviewer 2: Recordings are still videos, right? I would word it as "Images, video and audio files, educational video games ..."

> The media content type represents an easier digestible format of presenting information.

Reviewer: ```suggestion
The media content type represents an easily digestible format for presenting information.
```

> As such, it is typically used in conjunction with other content types, such as reading, slides or demos.

Reviewer: Elaborate on this that media items are typically embedded within the text destined for reading, practice items, assignments or slides.

Author: I did not elaborate on purpose because I was thinking this can be better explained at the "Teaching stages chapter". Of course we can elaborate on both sections

Reviewer: Does it have to do with teaching? It's about where the media items fit in, which is still part of the stored content.

When developing `media` items, it is important to follow the guidelines below:

> - use an open source or free development tool.

Author: I tried to keep it as separated from actual technology as possible because this falls under the infrastructure/open source tools chapter.

> - use an open source or free development tool.

Reviewer 3: I agree with @teodutu, providing some examples makes it easier for people to develop materials. Also, most of them are intuitive enough to be referrenced without a description.

> - use an open source or free development tool.

Reviewer 2: Agree with @teodutu and @Alex-deVis. Make it concrete, give the reader assistance. Examples are always a good idea.

> - store both the output (the actual image, video etc.) and the editable building blocks.

Reviewer: Not necessarily true. `.svg` images may serve as both the output and the editable building blocks (without requiring the `.drawio` files which are mostly the same thing).

Secondly, add a line about recordings and screenshots. There's nothing editable about them and unfortunately, they are to be stored as they are.

Author: >Not necessarily true. .svg images may serve as both the output and the editable building blocks (without requiring the .drawio files which are mostly the same thing).

That's just a particular case where the output and the editable building block are the same file.

Author: > Secondly, add a line about recordings and screenshots. There's nothing editable about them and unfortunately, they are to be stored as they are.

I will elaborate on this.

Reviewer 2: I would word it as "being contributable / editable" - where possible (screenshots can't fit this). Such that someone could then get the media content and be able to improve it according to the 5R content properties. Only then, I would mention storing both the original content and the output (where possible).

> 

Reviewer 3: It is worth mentioning that slides are better than gifs when it comes to illustrating processes or sequential actions. Slides enable students to go at their own pace and also require no additional infrastructure.

### Media

The `media` content type refers to audio, visual or audiovisual educational materials.
Images, videos, recordings, educational video games represent examples of such content.

The media content type represents an easily digestible format for presenting information.
As such, media is typically embbeded within the text that is destined reading, slides, demos, practice items and projects.

> When developing `media` items, it is important to follow the guidelines below:

Reviewer 2: Why is media marked with backticks here but not elsewhere? I would remove backticks.

- use an open source or free development tool.
- store both the output (the actual image, video etc.) and the editable building blocks.

#### Images

> For the creation of images and diagrams we use [draw.io](draw.io).

Reviewer 2: ```suggestion
For the creation of images and diagrams we use [draw.io](https://draw.io).
```

> `draw.io` is a free, easy to use tool which is available in the browser.

Reviewer 2: Argue why having it as a browser helps, that it exports SVG and that it contains many useful icons / basic blocks. Hence, our decision to use it.

The resulting images can be exported in different formats, such as PDF, PNG or SVG.

> When storing images, it is important that both the source file (with the extension `drawio` or `svg`) and the immutable output (PDF, PNG, JPG) are stored.

Reviewer 2: Mention why this is important. In general, argue our decisions.

Note that other tools for the generation of images may be used as long as they are free and provide the source file.

> 

Reviewer 2: Mention about screenshots.

> #### Audio-video recordings

Reviewer 2: ```suggestion
#### Audio-video Recordings
```

> Audio or video content is by its nature hard to modify.

Reviewer 2: I would do a classification of audio-video content and correlate it with use cases. Otherwise, it simply appears as a listing of potential approaches, but the reader wouldn't know what to use where.

I would add:
- Animations: succession of images that show a process, a set of steps - these can be created with an image software and then an image-tying software such as ffmpeg
- Screencasts: someone records their screen, it requires a screen recording software - give examples, this typically screencasts applications (terminal, browser, simulators)
- Actual recordings: a live recording with a camera of someone giving a talk or detailing a concept; this can also be, in certain cases, an audio recording only

I would then add sections on each of those and detail the actual methodology for producing these types of contents, together with sample tools they would use for them.

Reviewer 2: Either here, when presenting the media content types, or in each section, provide use cases for each of those. The reader should have answers to questions:
- What should I do if I want to do X (with respect to media content)? (get from use case to tool)
- What should I use this media content type for? (get from tool to use case)

Typically, if a video/audio recording requires alterations, it is re-recorded.
As such, the video/audio recording is stored as is and if modifications are required, it needs to be recreated.
In terms of the tools that can be used to create such recordings, given the variety of options, we do not give any specific recommendations.
Any device that can record at a decent quality is fine.

There is a particular type of video that can be created without using a camera.
By putting multiple similar images in a slideshow and automating the presentation, the video effect can be achieved.
This sort of video is useful for educational material that presents processes or phenomenons step by step.
To create such videos, we need to first create the images and then bundle them up using [ffmpeg](https://ffmpeg.org/).

> In this situation the image source files, the actual images and the video need to be stored in the content repository.

Reviewer 2: I don't think we are storing the video in the repository, do we? @teodutu ?

> #### Interactive applications

Reviewer 2: ```suggestion
#### Interactive Applications
```

Reviewer 2: Not sure if this should be part of the media section. This is more of "support software". But, then again, I can't think of a better place to add this content type.

Some educational scenarios involve the learner to interact with a specific application.
The application can be a video game, a web page or even a simple computer program that simulates a given context.

> For such situations, the code of the application should be stored alongside the information regarding the tools that are necessary to build the application.

Reviewer 2: This I don't get. You mean the code that's used by the application to run the demo / scenario, right? Not the application code. I would reword it to make it more clear.
